### DISCUSSION TITLE
How to specify name of generated dart class for view defined in .drift file

### DISCUSSION THREAD

**Discussion by Adam-Langley** [simolus3/drift#2960] [General]
I plan to use a convention where by my db entity classes are suffixed with "Entity", but the SQL tables themselves don't need that suffix (it's obvious they're a table :)
I know I can control the name of the generated SQL Table using annotations - however - for my views, I have those defined in .drift files - how do I override how drift names the view classes, so I can add the suffix, independently from the name of the SQL view?

i.e

dart class: CustomerEntity
sql table: customer

sql view (generated by .drift file): customers_with_address
dart class (generated by drift build): CustomersWithAddressEntity

**Comment by frankvollebregt** [comment]
Interesting, for tables defined in `.drift` files, this is pretty simple
```sql
CREATE TABLE items (
  id INT NOT NULL PRIMARY KEY,
  description TEXT
) AS ItemEntity;
```
But I suspect something like this does not (yet) exist for `VIEW`s. The syntax would be a bit confusing since `AS` is already used when defining a view, and I can't really find anything about in in the docs.

Something that _is_ certainly possible right now, is to define your `CustomerWithAddressEntity` class in Dart code and use the `WITH` keyword to map results of your view to this custom class, as per [the second example here](https://drift.simonbinder.eu/docs/advanced-features/custom_row_classes/#existing-row-classes-in-drift-files). I.e.
```sql
CREATE VIEW my_view WITH ExistingClass AS SELECT ...
```
But of course this is a hassle to maintain. I've tried a few variants using the `AS` keyword, but could not get it to work. I think @simolus3 will probably be able to shed some light on this :)

## Ideas (for a possible implementation of this as a new feature)
Assuming it does indeed not exist yet, I had the following ideas trying to get it working (i.e. these might make sense in the syntax used throughout Drift files):
1. Simple extra `AS` after the name: a bit confusing because of the double `AS` but most similar to the use of `WITH` in views
```sql
CREATE VIEW my_view AS my_view_entity AS SELECT * FROM customers;
```
2. Use brackets to clarify it's a name alias. I like this more, even though it changes the syntax a bit from the `WITH` syntax.
```sql
CREATE VIEW (my_view AS my_view_entity) AS SELECT * FROM customers;
```
3. Place `AS` in the end, like with tables, which suffers from the same confusing syntax as 1. but is also not consistent with the `WITH` syntax
```sql
CREATE VIEW my_view AS SELECT * FROM customers AS my_view_entity;
```

**Reply by simolus3** [reply]
> I've tried a few variants using the `AS` keyword, but could not get it to work.

That's because it really is unsupported at the moment. The reason for that is that `AS` would kind of conflict with the `AS` from the view definition.

> Place `AS` in the end, like with tables, which suffers from the same confusing syntax as 1. but is also not consistent with the `WITH` syntax

I've considered this, but it makes the syntax conflict much worse because the `AS my_view_entity` should really be part of the `FROM customers` (giving the table an alias).

It took some parsing tricks and the changes have not yet been released, but you will be able to use the syntax from option 1 in the next drift version (so `CREATE VIEW my_view AS my_view_entity AS SELECT ...`).

**Reply by Adam-Langley** [reply]
Hi @simolus3 (and thanks for your input @frankvollebregt )
Wow, how serendipitous!

I will patiently await the release.

Thanks again,

**Reply by simolus3** [reply]
I've just published drift 2.17 which includes these changes.
